parser/src/lexer.rs

```diff
@@ -1669,6 +1681,85 @@ mod tests {
         )
     }
 
+    #[test]
+    fn test_jupyter_magic_assignment() {
+        let source = r"
+pwd = !pwd
+foo = %timeit a = b
+bar = %timeit a % 3
+baz = %matplotlib \
+        inline"
```

Comment:
All of the valid positions for magic command token in terms of an assignment statement.